### PR TITLE
Centralize some shared code related to finding email address of authorized user making a request

### PR DESCRIPTION
**Pull request by netsettler**
There's a bit of code that's duplicated between `types/cohort.py` and `types/family.py` that I wanted to centralize so that I can maybe use it for other purposes.

Opportunistic:  The code for `generate_fastq_file` was not yet used and is now available via `dcicutils.data_utils.generate_sample_fastq_file`, so I removed it here. That makes the diffs of `utils.py` a little messy because I'm replacing one function with another when they are unrelated because it happens to be the same place in the file. Side-by-side diff will probably clear that up.

Adding @sbreiff  and @KorayKirli  as reviewers since this is code they've touched in the past. But I'm not really changing the functionality, just consolidating, so I don't expect this will upset anything.

### CODE REVIEW

File: src/encoded/util.py [hunk 1]
```diff
@@ -238,38 +238,35 @@ def create_empty_s3_file(s3_client, bucket: str, key: str):
...
-        content += '@SEQUENCE{} length={}\n'.format(i, length)
-        content += ''.join(random.choice(bases) for i in range(length)) + '\n'
-        content += '+\n'
-        content += 'I' * length + '\n'
-    with gzip.open(filename, 'w') as outfile:
-        outfile.write(content.encode('ascii'))
-
-    return filename
+    try:
+        # import pdb;pdb.set_trace()
```
Comment (willronchetti) [inline src/encoded/util.py:251]: Remove

**Comment by netsettler** [comment]
yeah, unit testing would be good. but one reason to do this is that it didn't have unit testing before other than being used in something bigger. by separating things, the need and ability to test becomes clearer.

**Comment by netsettler** [comment]
i don't know, though, that i have time to do it today so probably will not include it here. i'll be watching for a moment when i can do that in the future.

**Comment by netsettler** [comment]
I see the workbook test is still failing intermittently but will ignore that as not relevant here.

**Pull request merged**